app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java

```diff
@@ -0,0 +1,158 @@
+
+  @BindView(R.id.detail_sun) ForecastDetailCompoundView detailSun;
+  @BindView(R.id.detail_wind) ForecastDetailCompoundView detailWind;
+  @BindView(R.id.detail_temperature) ForecastDetailCompoundView detailTemperature;
+
+  HomeContract.Presenter presenter;
+
+  @Override protected int getLayoutId() {
+    return R.layout.activity_home;
+  }
+
+  @Override protected void onCreate(Bundle savedInstanceState) {
+    //override theme defined in the xml for splash screen effect
+    setTheme(R.style.AppTheme);
+
+    super.onCreate(savedInstanceState);
+
+    initToolbar(toolbar);
+    showTitleBar(false);
+
+    setNavigationView();
+
+    presenter = new HomePresenterImpl(this);
+    presenter.fetchWeatherDetails();
+  }
+
+  @Override public boolean onCreateOptionsMenu(Menu menu) {
+    MenuInflater inflater = getMenuInflater();
+    inflater.inflate(R.menu.menu_main, menu);
+    return true;
+  }
+
+  @Override public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case android.R.id.home:
+        drawerLayout.openDrawer(GravityCompat.START);
+        return true;
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+  private void setNavigationView() {
+    navigationView.setNavigationItemSelectedListener(menuItem -> {
+      //TODO
+      Toast.makeText(HomeActivity.this, menuItem.toString(), Toast.LENGTH_SHORT).show();
+
+      drawerLayout.closeDrawers();
+
+      // Add code here to update the UI based on the item selected
+      // For example, swap UI fragments here
+      return true;
+    });
+  }
+
+  @Override public Context getContext() {
+    return this;
+  }
+
+  @Override public void showProgressDialog(String message, boolean cancelable) {
+    getProgressDialog().setMessage(message);
+    getProgressDialog().setCancelable(cancelable);
+    getProgressDialog().show();
+  }
+
+  @Override public void hideProgressDialog() {
+    getProgressDialog().dismiss();
+  }
+
+  @Override public void setCurrentForecast(double latitude, double longitude,
+      CurrentForecast currentForecast) {
+    tvLocation.setText(LocationUtils.getLocality(latitude, longitude, this));
+    tvCurrentForecastSummary.setText(currentForecast.getSummary());
+    tvTempCurrent.setText(String.valueOf(Math.round(currentForecast.getTemperature())));
+    ivWeather.setImageResource(
+        WeatherImageMapper.getImageResource(currentForecast.getIcon()));
+  }
+
+  @Override public void setDailyForeCast(List<DailyForecast.DailyData> dailyForecastList) {
+    for (DailyForecast.DailyData dailyData : dailyForecastList) {
+      addDailyForecastView(dailyData);
+    }
+
+    DailyForecast.DailyData forecastDetailToday = dailyForecastList.get(0);
+
+    setForecastDetails(forecastDetailToday);
+  }
+
+  private void addDailyForecastView(DailyForecast.DailyData dailyData) {
+    String averageTemperature = String.valueOf(Math.round(
+        MathUtils.getAverage(dailyData.getTemperatureHigh(),
```

Comment:
MathUtils.getAverage should use variable arguments so that any number of temperature can be sent

Reply:
:ok: